src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,774 @@
+        /// </summary>
+        private async Task GetChangesAsync(CancellationToken token)
+        {
+            try
+            {
+                using (var connection = new SqlConnection(this._connectionString))
+                {
+                    await connection.OpenAsync(token);
+
+                    using (SqlTransaction transaction = connection.BeginTransaction(System.Data.IsolationLevel.RepeatableRead))
+                    {
+                        try
+                        {
+                            // Update the version number stored in the global state table if necessary before using it.
+                            using (SqlCommand updateTablesPreInvocationCommand = this.BuildUpdateTablesPreInvocation(connection, transaction))
+                            {
+                                await updateTablesPreInvocationCommand.ExecuteNonQueryAsync(token);
+                            }
+
+                            // Use the version number to query for new changes.
+                            using (SqlCommand getChangesCommand = this.BuildGetChangesCommand(connection, transaction))
+                            {
+                                var rows = new List<IReadOnlyDictionary<string, string>>();
+                                using (SqlDataReader reader = await getChangesCommand.ExecuteReaderAsync(token))
+                                {
+                                    while (await reader.ReadAsync(token))
+                                    {
+                                        rows.Add(SqlBindingUtilities.BuildDictionaryFromSqlRow(reader));
+                                    }
+                                }
+
+                                this._rows = rows;
+                            }
+
+                            // If changes were found, acquire leases on them.
+                            if (this._rows.Count > 0)
+                            {
+                                using (SqlCommand acquireLeasesCommand = this.BuildAcquireLeasesCommand(connection, transaction))
+                                {
+                                    await acquireLeasesCommand.ExecuteNonQueryAsync(token);
+                                }
+                            }
+                            transaction.Commit();
+                        }
+                        catch (Exception ex)
+                        {
+                            this._logger.LogError("Commit Exception Type: {0}", ex.GetType());
+                            this._logger.LogError("  Message: {0}", ex.Message);
+                            try
+                            {
+                                transaction.Rollback();
+                            }
+                            catch (Exception ex2)
+                            {
+                                this._logger.LogError("Rollback Exception Type: {0}", ex2.GetType());
```

Comment:
Same here, better message would be something like `Exception rolling back transaction after error occurred getting list of changes for table...`

Reply:
I haven't added the 'after error occurred ...' part as that would already be logged in preceding log message.